cluster/pre_validation.sls

```diff
@@ -15,8 +15,12 @@
   {% endif %}
 {% endif %}
 
-{# Check HA exporter #}
-{% if cluster.add_exporter is defined and cluster.add_exporter == false %}
-  {% do cluster.pop('ha_exporter') %}
+{% if cluster.sbd.configure_sbd_checkbox is not defined or cluster.sbd.configure_sbd_checkbox is sameas false %}
+{% do cluster.sbd.pop('configure_resource', none) %}
 {% endif %}
-{# Check HA exporter finish #}
+
+{# Check monitoring via HA exporter #}
+{% if cluster.monitoring_enabled is defined and cluster.monitoring_enabled == false %}
+  {% do cluster.pop('monitoring_enabled') %}
```

Comment:
Actually, you don't need to do any of this. I think it works without this condition.
The `monitoring_enabled` is set in our `defaults.yml` so it will be used always, so we don't care if it's defined or not

Reply:
Its strange, the monitoring states do not run if I disable the entry in form. The `cluster.pop` must be working for simple types like `$type: boolean` here